src/core/interface.scala

```diff
@@ -61,7 +61,8 @@ case class SealedTrait[Typeclass[_], Type]
                       (typeInfo: TypeInfo,
                        subtypes: IArray[SealedTrait.Subtype[Typeclass, Type, _]],
                        annotations: IArray[Any],
-                       typeAnnotations: IArray[Any]) extends Serializable:
+                       typeAnnotations: IArray[Any],
+                       isEnum: Boolean = false) extends Serializable:
```

Comment:
The `SealedTrait` name can be misleading, it can represent both the sealed hierarchies of `sealed trait`s as well as `enum`s. To implement this issue, a small macro would be needed similar to `isObject` here: https://github.com/softwaremill/magnolia/blob/d8be357042cbf49277fbbb0dfaa4792bbef1f348/src/core/macro.scala#L22
All flags in Quotes API can be found [here](https://github.com/lampepfl/dotty/blob/6655587a00ae72c300ea1d4a1e6966a8aff80ae7/compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala#L2656).
This macro should be used in `magnolia.scala` when creating `SealedTrait` instances.

Reply:
Thanks for help